src/terraform_module_generator/schema/blocks/block.py

```diff
@@ -0,0 +1,230 @@
+"""
+    :copyright: 2022 Inmanta
+    :contact: code@inmanta.com
+    :license: Inmanta EULA
+"""
+import textwrap
+import typing
+
+from inmanta_module_factory import builder, inmanta
+from inmanta_module_factory.helpers.utils import inmanta_entity_name, inmanta_safe_name
+
+from terraform_module_generator.schema import const, mocks
+from terraform_module_generator.schema.attributes.base import Attribute
+from terraform_module_generator.schema.attributes.collection import CollectionAttribute
+from terraform_module_generator.schema.attributes.structure import StructureAttribute
+from terraform_module_generator.schema.helpers.cache import cache_method_result
+from terraform_module_generator.schema.mocks.nested_block import NestedBlockMock
+
+if typing.TYPE_CHECKING:
+    from terraform_module_generator.schema.blocks.nested_block import NestedBlock
+
+
+class Block:
+    def __init__(self, name: str, path: typing.List[str], schema: typing.Any) -> None:
+        self.name = name
+        self.path = path
+
+        nested_blocks: typing.List[mocks.NestedBlockMock] = []
+        self.attributes: typing.List[Attribute] = []
+        for attribute in Block.get_attributes(path + [inmanta_safe_name(name)], schema):
+            if isinstance(attribute, StructureAttribute):
+                nested_blocks.append(
+                    mocks.NestedBlockMock(
+                        type_name=attribute.name,
+                        block=attribute.block_mock(),
+                        nesting=1,  # SINGLE
+                        min_items=0 if attribute.optional else 1,
+                        max_items=1,
+                    )
+                )
+                continue
+
+            if not isinstance(attribute, CollectionAttribute):
+                self.attributes.append(attribute)
+                continue
+
+            if not isinstance(attribute.inner_type, StructureAttribute):
+                self.attributes.append(attribute)
+                continue
+
+            nested_blocks.append(attribute.nested_block_mock())
```

Comment:
Attribute doesn't have this method, so I wonder: doesn't mypy complain? 

and second, it would be good to have the base method, so there is a place to document it.